File: 151001/Hlebanova/LR2/LR2/Mappers/CreatorMapper.cs
using Riok.Mapperly.Abstractions;
using LR2.Dto.Request.CreateTo;
using LR2.Dto.Response;
using LR2.Models;

namespace LR2.Mappers;

[Mapper]
public static partial class CreatorMapper
{
    public static partial Creator ToEntity(this CreateCreatorRequestTo createCreatorRequestTo);
    public static partial CreatorResponseTo ToResponse(this Creator creator);
    public static partial Creator Map(CreateCreatorRequestTo createCreatorRequestTo);
    public static partial CreatorResponseTo Map(Creator creator);
    public static partial IEnumerable<CreatorResponseTo> Map(IEnumerable<Creator> creators);
}

File: 151003/Patiyuk/TaskRedis/publisher-service/src/main/java/by/bsuir/publisherservice/mapper/CreatorMapper.java
package by.bsuir.publisherservice.mapper;

import by.bsuir.publisherservice.dto.request.CreatorRequestTo;
import by.bsuir.publisherservice.dto.response.CreatorResponseTo;
import by.bsuir.publisherservice.entity.Creator;
import org.mapstruct.Mapper;
import org.mapstruct.Mapping;
import org.mapstruct.MappingTarget;

@Mapper(componentModel = "spring")
public interface CreatorMapper {

    @Mapping(target = "id", ignore = true)
    @Mapping(target = "firstName", source = "firstname")
    @Mapping(target = "lastName", source = "lastname")
    Creator toEntity(CreatorRequestTo request);

    @Mapping(target = "firstname", source = "firstName")
    @Mapping(target = "lastname", source = "lastName")
    CreatorResponseTo toResponseTo(Creator entity);

    @Mapping(target = "firstName", source = "firstname")
    @Mapping(target = "lastName", source = "lastname")
    Creator updateEntity(@MappingTarget Creator entity, CreatorRequestTo request);
}
